packages/shared/src/hooks/vote/types.ts

```diff
@@ -0,0 +1,78 @@
+import { Post, ReadHistoryPost, UserPostVote } from '../../graphql/posts';
+import { Origin } from '../../lib/analytics';
+import { PostAnalyticsEventFnOptions } from '../../lib/feed';
+import { UseMutationMatcher } from '../mutationSubscription/types';
+
+export type ToggleVoteProps = {
+  origin: Origin;
+  post: Post | ReadHistoryPost;
+  opts?: PostAnalyticsEventFnOptions;
+};
+
+export type VoteProps = {
+  id: string;
+};
+
+export type UseVotePost = {
```

Comment:
More of a generic question, why do we make a distinguisment between `upvote` and `toggleUpvote` why not one uniform way?

Reply:
Because in some places we still need base methods. Companion has a little different implementation and I did not want to break it up in this go. We also might want to just trigger mutation without toggle logic. I think it should be useful still.

Reply:
Ah ok, let's rather refactor then down the line.
Don't want to increate scope, was just curious where we differentiate.